src/main.rs

```diff
@@ -160,12 +153,42 @@ fn start_program(sess: &mut Session, elf: &Elf) -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+/// Set rtt to blocking mode
+fn set_rtt_to_blocking(
+    core: &mut Core,
+    main_fn_address: u32,
+    rtt_buffer_address: u32,
+) -> anyhow::Result<()> {
+    // set and wait for a hardware breakpoint at the beginning of `fn main()`
+    core.set_hw_breakpoint(main_fn_address)?;
+    core.run()?;
+    core.wait_for_core_halted(Duration::from_secs(5))?;
+
+    // calculate address of up-channel-flags inside the rtt control block
+    const OFFSET: u32 = 44;
+    let rtt_buffer_address = rtt_buffer_address + OFFSET;
```

Comment:
Not saying that there would be a big gain, if at all, but from a "performance" perspective we probably should calculate the `rtt_buffer_address` in between `core.run` and `core.wait_for_core_halted`, shouldn't we? Since else we idle until we hit the breakpoint anyways.

Reply:
No, it's the compiler's job to reorder code to make it faster